From 7c6b7c7b37ab37e1faba42865a502d663e985d13 Mon Sep 17 00:00:00 2001
From: "Song.Li" <Song.Li@windriver.com>
Date: Sat, 17 Nov 2012 12:41:34 +0800
Subject: [PATCH] remove mysql detective from configure.in

mysql-connector-odbc use mysql_config command to get mysql lib and inc flags.
this is unsuitable for cross-compiling, so remove this and pass the right
lib and inc flags by command line.

Signed-off-by: Song.Li <Song.Li@windriver.com>
---
 configure.in |   75 +---------------------------------------------------------
 1 file changed, 1 insertion(+), 74 deletions(-)

diff --git a/configure.in b/configure.in
index 93ad7c5..7f34488 100644
--- a/configure.in
+++ b/configure.in
@@ -130,83 +130,10 @@ AC_CHECK_FUNCS(localtime_r)
 #                                                                 #
 ###################################################################
 
-AC_ARG_WITH(mysql-path,
-            [  --with-mysql-path=DIR   Path where MySQL is installed],
-            mysql_path=$withval,
-            mysql_path="")
-
-mysql_config=""
-building_against_source=""
-MYSQL_PATH_ARG=""
-
-if test "$mysql_path" != ""
-then
-  MYSQL_PATH_ARG="--with-mysql-path=$mysql_path"
-  if test -f $mysql_path/libmysql_r/libmysqlclient_r.la
-  then
-    building_against_source="true"
-    CFLAGS="$CFLAGS -I$mysql_path/include"
-    MYSQL_LIB="$mysql_path/libmysql_r/libmysqlclient_r.la"
-  else
-    mysql_config="$mysql_path/bin/mysql_config"
-    if test ! -x $mysql_config
-    then
-      AC_MSG_ERROR([File $mysql_config doesn't exists or isn't executable.  Please specify with --with-mysql-path where bin/mysql_config can be found]);
-    fi
-  fi
-else
-  # Search where mysql_config is installed
-  TEST_PATHS="/opt/mysql/mysql /usr/local/mysql /usr"
-  for i in $TEST_PATHS
-  do
-    mysql_config="$i/bin/mysql_config"
-    if test -x $mysql_config
-    then
-      MYSQL_PATH_ARG="--with-mysql-path=$TEST_PATHS"
-      break;
-    fi   
-    mysql_config=""
-  done
-
-  if test "$mysql_config" = ""
-  then
-    AC_MSG_ERROR([Can't find mysql_config in $TEST_PATHS]);
-  fi
-fi
+CFLAGS="$CFLAGS $MYSQL_CFLAGS"
 
 AC_SUBST(MYSQL_PATH_ARG)
 
-if test "x$building_against_source" = x
-then
-  OPT=`$mysql_config --cflags`
-  if test $? != "0"
-  then
-    AC_MSG_ERROR([Could not execute $mysql_config. Please check your installation])
-  fi
-
-  # We have to remove any ' around paths as this confuses configure
-  OPT=`echo $OPT | sed -e "s;';;g"`
-  CFLAGS="$OPT $CFLAGS"
-
-  OPT=`$mysql_config --libs_r`
-  if test $? != "0"
-  then
-    # mysql_config that is before 4.0.17
-    # In this case assume we can use the same compile options for the
-    # thread safe library as for the normal
-    OPT=`$mysql_config --libs | sed -e 's;lmysqlclient;lmysqlclient_r;'`
-  else
-    OPT=`echo $OPT | sed -e "s;';;g"`
-  fi
-  MYSQL_LIB="$OPT"
-
-  SAVE_LIBS=$LIBS
-  LIBS="$LIBS $MYSQL_LIB"
-  AC_CHECK_FUNC(mysql_real_query,[],
-  [ AC_MSG_ERROR([Could not use the single thread MySQL client library, please check config.log for errors]) ])
-  LIBS=$SAVE_LIBS
-fi
-
 AC_SUBST(MYSQL_LIB)
 
 ###################################################################
-- 
1.7.9.5

